nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/fastqc.nf

```diff
@@ -10,14 +10,24 @@ process fastqc {
                 }
 
     input:
-    set val(name), file(reads) from ch_read_files_fastqc
+    tuple val(name), val(single_end), path(reads)
 
     output:
-    file "*_fastqc.{zip,html}" into ch_fastqc_results
+    path "*.{zip,html}"
 
     script:
-    """
-    fastqc --quiet --threads $task.cpus $reads
-    """
+    // Add soft-links to original FastQs for consistent naming in pipeline
+    if (single_end) {
```

Comment:
Where does this variable come from?

Reply:
The `single_end` value is auto-detected in  `check_samplesheet.py` and outputted into the channels to be used as a variable in the pipeline as opposed to a parameter e.g. [here](https://github.com/drpatelh/tools/blob/b8f4caf723854ca95692157461b1f34b305db5fb/nf_core/pipeline-template/%7B%7Bcookiecutter.name_noslash%7D%7D/modules/local/check_samplesheet.nf#L29). This allows for flexibility in using both SE/PE reads in the same run of the pipeline.

Reply:
Right, but that's not in the scope of this process is it?

Reply:
Yep, it is. Its defined as `val(single_end)` in the `input` section for the process.

Reply:
Ah nice 👍  Sorry, that's what I was curious about.. I was worried it was a copy + paste error and that it was left over from something else..